combine_PCA_WDF.py

```diff
@@ -0,0 +1,185 @@
+All of the abovementioned scripts should be in your working directory (maybe you need to add the __init__.py file in the same folder as well.
+You should first choose the data file with the map scan in the .wdf format (I could add later the input dialog)
+You set the "snake" variable to "True" or "False" (depending on the way the map was recorded)
+You choose the number of components. 
+That's it
+You should first get the plot of all the components found,
+Then the heatmap of the mixing coefficients: when you double-click on a pixel on this map, 
+it will pop-up another plot with the spectra recorded at this point, together with the contributions of each component
+'''
+
+'''Ideas to improve the script:
+    Add the possiblity to plot the heatmap of the max_height of one selected peak (region) or it's area
+    
+    I should test Katia's idea, which is to do the deconvolultion on several subensembles of spectra,
+    then on the whole spectra and then compare if the add up.
+    
+    Check out the issue of (non)normalized mixture coefficients.
+    Would it be more meaningfull to renormalize the components (divide by max value, for exemple),
+    so the mixture coefficients should add up to 1.
+    (how would the division by max value reflect to mixture coeffs? - should they be then multiplied by the same value?)
+    Now, each component has very different absolute values, 
+    so even if the mixture coeff is small, the first component is nevertheless predominant
+    in the decomosed spectra by the sheer fact that it's intensity is several orders of magnitude greater
+    than that of some other components
+    
+    
+'''
+
+#filename = 'Data/Test-Na-SiO2 0079 -532nm-obj100-p100-10s over night carto.wdf'#scan_type 1, measurement_type 3
+#filename = 'Data/Test-Na-SiO2 0079 droplet on quartz -532nm-obj50-p50-15s over night_Copy_Copy.wdf'#scan_type 2, measurement_type 2
+#filename = 'Data/Test quartz substrate -532nm-obj100-p100-10s.wdf'#scan_type 2, measurement_type 1
+#filename = 'Data/Hamza-Na-SiO2-532nm-obj100-p100-10s-extended-cartography - 1 accumulations.wdf'#scan_type 2 (wtf?), measurement_type 3
+#filename = 'Data/M1SCMap_2_MJ_Truncated_CR2_NF50_PCA3_Clean2_.wdf'
+#filename = 'Data/M1ANMap_Depth_2mm_.wdf'
+filename = 'Data/M1SCMap_depth_.wdf'
+snake = True # The scanning mode: either always from left to right (snake = False), either left->right right->left (snake = True)
+n_components=5
+# one should always check if the spectra were recorded with the dead pixels included or not
+# It turns out that firs 10 and the last 16 pixels on the Renishaw SVI spectrometer detector are reserved, 
+# and no signal is recorded on those pixels by the detector. So we should either enter these parameters inside the Wire settings
+# or if it's not done, remove those pixels here manually
+# Furthermore, we sometimes want to perform the deconvolution only on a part of the spectra, so here you define the part that interests you
+ssss = slice(575,950)
```

Comment:
Please make sure that all variable names are as informative as possible. E.g _slice, spectra, reader... (a lot are already fine like laser_wavelength etc.)